guava/src/com/google/common/reflect/ClassPath.java

```diff
@@ -478,10 +484,18 @@ protected void scanJarFile(ClassLoader classloader, JarFile file) {
 
     @Override
     protected void scanDirectory(ClassLoader classloader, File directory) throws IOException {
-      scanDirectory(directory, classloader, "");
+      scanDirectory(directory, classloader, "", new HashSet<>());
     }
 
-    private void scanDirectory(File directory, ClassLoader classloader, String packagePrefix)
+    /**
+     * Recursively scan the given directory, adding resources for each file encountered.  Symlinks which have already
```

Comment:
From my past experience, the Guava team prefer sentences be separated by one space only. I'd suggest running google-java-format on this file too, as that would (or at least should) fix the javadoc to meet their standards.